src/vc/vc.spec.ts

```diff
@@ -386,4 +386,113 @@ describe('VC Resolver', function () {
       expect(vcRevokeStatusNew).to.be.false;
     });
   });
+
+  describe('When creating a private/sharable VC', async () => {
+    // Test for after sharable VCs have been implemented.
+
+
+    it('allows the current runtime to use generate key using vc api', async () => {
+      const vcKey = await runtime.vc.generateKey();
+      expect(vcKey).to.be.string;
+    });
+
+    it('does not allow the current runtime to generate same keys using vc api', async () => {
+      const vcKey = await runtime.vc.generateKey();
+      expect(vcKey).to.be.string;
+
+      const vcKey2 = await runtime.vc.generateKey();
+      expect(vcKey2).to.be.string;
+      expect(vcKey).to.be.not.eq(vcKey2);
+    });
+
+    it('allows to store and retrieve a sharable vc from the same account', async () => {
+      const vcKey = await runtime.vc.generateKey();
+      encryptionInfo = { key: vcKey };
+      expect(vcKey).to.be.string;
+
+      const storedVcDoc = await runtime.vc.storeVc(minimalValidVcData,
+        encryptionInfo);
+      const fetchedVcDoc = await runtime.vc.getVc(storedVcDoc.id.replace('vc:evan:testcore:', ''),
+        encryptionInfo);
+
+      expect(storedVcDoc).to.be.deep.eq(fetchedVcDoc);
+    });
+
+    it('does not allow to retrieve a sharable vc with wrong key', async () => {
+      const vcKey = await runtime.vc.generateKey();
+      const vcKey2 = await runtime.vc.generateKey();
+      expect(vcKey).to.be.string;
+
+      encryptionInfo = { key: vcKey };
+      const storedVcDoc = await runtime.vc.storeVc(minimalValidVcData,
+        encryptionInfo);
+      encryptionInfo = { key: vcKey2 };
+      const fetchedVcDoc = runtime.vc.getVc(storedVcDoc.id.replace('vc:evan:testcore:', ''),
+        encryptionInfo);
+
+      await expect(Error).to.exist;
```

Comment:
Does this actually do anything? I think this always resolves to true, no?

Reply:
In this basically I generate 2 different keys. I store the VC with key1 and in then I change the key in the encryptionInfo to key2 and try to retrieve the VC using key2. There should be an Error thrown as the key is different though the runtime is same. So yes it will resolve true in case of Error.

Reply:
Ok you were correct. the Error is a class so it will always return true. My bad got a bit confused.